deltachat-ios/Chat/ChatViewController.swift

```diff
@@ -2411,6 +2411,11 @@ extension ChatViewController: BaseMessageCellDelegate {
         if Utils.isEmail(url: url) {
             let email = Utils.getEmailFrom(url)
             self.askToChatWith(email: email)
+        } else if url.isDeltaChatInvitation,
+                  let qrCode = Utils.makeDeltaChatInvitationQRCode(from: url),
+                  let appDelegate = UIApplication.shared.delegate as? AppDelegate,
+                  let appCoordinator = appDelegate.appCoordinator {
+            appCoordinator.handleQRCode(qrCode)
```

Comment:
when hitting "Cancel" in the question, the keyboard will be gone:

https://github.com/user-attachments/assets/f8e1e7af-027d-48ab-89b7-f7a7d5032d7d

Reply:
The first thing that comes to mind is to inject for an optional closure to `handleQRCode` of `QrPageController` what to do when tapping the cancel-button.
This also means forwarding it everywhere